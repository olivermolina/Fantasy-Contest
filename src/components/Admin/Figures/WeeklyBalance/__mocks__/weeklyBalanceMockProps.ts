export const weeklyBalanceMockData = [
  {
    agent: null,
    id: '1',
    username: 'jdoe',
    firstname: 'John',
    lastname: 'Doe',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
  {
    agent: null,
    id: '2',
    username: 'jwick',
    firstname: 'John',
    lastname: 'Wick',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
];
