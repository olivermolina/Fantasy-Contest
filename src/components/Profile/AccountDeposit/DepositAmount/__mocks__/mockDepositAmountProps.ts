export const mockDepositAmountProps = {
  setDepositAmount: () => {
    alert('Set Deposit amount');
  },
  handleNext: () => {
    alert('handleNext');
  },
  depositAmount: 10,
  isFirstDeposit: false,
  maxMatchDeposit: 50,
  depositAmountOptions: [10, 25, 50, 75, 100, 125],
  reloadBonusAmount: 0,
  reloadBonusType: 'FLAT',
  showBonusCreditStyles: false,
  minDepositAmount: 10,
};
