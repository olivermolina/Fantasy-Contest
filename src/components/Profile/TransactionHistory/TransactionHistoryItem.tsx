import React from 'react';
import { TransactionHistoryInterface } from '~/server/routers/user/transactionHistory';
import classNames from 'classnames';
import dayjs from 'dayjs';
import { EntryDatetimeFormat } from '~/constants/EntryDatetimeFormat';

interface Props {
  row: TransactionHistoryInterface;
}

export default function TransactionHistoryItem(props: Props) {
  const { row } = props;

  return (
    <div
      className={
        'grid grid-cols-2 lg:grid-cols-6 gap-2 lg:gap-4 bg-inherit p-4'
      }
    >
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>
          Transaction Date
        </span>
        <span className={'text-white text-md lg:text-lg'}>
          {dayjs(row.transactionDate).format(EntryDatetimeFormat)}
        </span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Amount</span>
        <span className={'text-white text-md lg:text-lg'}>
          {row.amount.toFixed(2)}
        </span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Type</span>
        <span className={'text-white text-md lg:text-lg'}>{row.type}</span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>
          Free Credits
        </span>
        <span className={'text-white text-md lg:text-lg'}>
          {row.amountBonus.toFixed(2)}
        </span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Details</span>
        <span className={'text-white text-md lg:text-lg'}>{row.details}</span>
      </p>
      <p className={'flex flex-col gap-2'}>
        <span className={'text-lightText text-sm lg:text-md'}>Status</span>
        <div
          className={
            'flex w-fit text-md rounded-lg lg:text-lg bg-[#1A4265] items-center gap-2 p-2 '
          }
        >
          <span
            className={classNames('rounded-full p-2 w-2 h-2', {
              'bg-green-500': row.status === 'COMPLETE',
              'bg-yellow-600': row.status === 'PENDING',
              'bg-red-600': row.status === 'FAILED',
            })}
          />
          <p
            className={classNames('uppercase text-md text-white', {
              'text-green-500': row.status === 'COMPLETE',
              'text-yellow-600': row.status === 'PENDING',
              'text-red-600': row.status === 'FAILED',
            })}
          >
            {row.status}
          </p>
        </div>
      </p>
    </div>
  );
}
