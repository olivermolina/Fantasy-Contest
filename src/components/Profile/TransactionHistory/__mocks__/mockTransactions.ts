export const mockTransactions = [
  {
    id: 'ntTfe7gRlpn5jdFs',
    transactionDate: '2022-12-22T03:25:11.630Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'ethq8juazC7KtvgH',
    transactionDate: '2022-12-22T03:25:11.577Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'UMI9tDWaT7xNtVHg',
    transactionDate: '2022-12-22T01:03:59.711Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'wgbHYWZqSBtIctX8',
    transactionDate: '2022-12-21T05:27:23.935Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'ZscFw9kb0s2XmVEs',
    transactionDate: '2022-12-21T05:27:00.552Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'd7Vopc2W6Q9wGEDZ',
    transactionDate: '2022-12-21T05:26:28.270Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'xDy3kfbxy4cjD9qr',
    transactionDate: '2022-12-20T04:22:03.133Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'Qz3Xz6EkgYJ9KClm',
    transactionDate: '2022-12-20T02:35:19.720Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'Gfwsr6ZP7GX8mFWP',
    transactionDate: '2022-12-20T00:41:57.070Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'gLSDx0u0qogrxNti',
    transactionDate: '2022-12-19T13:32:25.934Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'NZJuTNNAnYQ4aG6U',
    transactionDate: '2022-12-19T13:31:58.326Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'cVPtkMyFhjYHXNiY',
    transactionDate: '2022-12-19T13:19:49.331Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'S5FenaRQOxWBG2vB',
    transactionDate: '2022-12-19T13:19:25.819Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'EsL5Xcf0zTUCDAFV',
    transactionDate: '2022-12-19T13:18:42.771Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'uKqxWbt5IRyku8XU',
    transactionDate: '2022-12-19T05:00:30.759Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '4WQpaYjn3syCcbT8',
    transactionDate: '2022-12-19T04:32:03.334Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'BJeZdXsf52cqaarp',
    transactionDate: '2022-12-19T01:19:32.357Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 's9ui5kXzl4Qar8Vw',
    transactionDate: '2022-12-19T01:19:20.223Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'GxR8VdbNKsNNDWsO',
    transactionDate: '2022-12-19T01:19:00.994Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'cxnTp3XM4VYkcrOv',
    transactionDate: '2022-12-18T22:25:20.572Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '1ZbLLVXDTK5qtNA0',
    transactionDate: '2022-12-18T18:24:48.904Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'jOgjBmWicUP7zgHY',
    transactionDate: '2022-12-18T04:35:21.404Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'NGgjxnx3KiRH0T1n',
    transactionDate: '2022-12-18T01:51:16.554Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'kVi5sXUOsAk2YjHw',
    transactionDate: '2022-12-17T22:07:08.562Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '2huiiABzKd233DUp',
    transactionDate: '2022-12-17T22:07:08.529Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'n9Bos4l02nMHL1X8',
    transactionDate: '2022-12-17T22:07:08.489Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'q9eNnyCqQynwliII',
    transactionDate: '2022-12-17T20:28:00.845Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'HrE8gpCArGvFu584',
    transactionDate: '2022-12-17T16:28:20.360Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'lT4KgvQIqy8Guvum',
    transactionDate: '2022-12-17T16:27:54.484Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '9CP5aBaUNFJ2Aacy',
    transactionDate: '2022-12-17T16:27:35.542Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'KvpwO3N2ezYMq8Gz',
    transactionDate: '2022-12-17T16:27:06.735Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'j4ZBcUkOgC0YHoWC',
    transactionDate: '2022-12-16T04:22:02.643Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'YgoQnQANhLHahJgZ',
    transactionDate: '2022-12-16T04:22:02.594Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'woxXFFUWNzR4r1GH',
    transactionDate: '2022-12-16T03:30:13.948Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'z0X8Plc592o8Z4nt',
    transactionDate: '2022-12-16T03:01:14.869Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'R28Opyqm6BiiHSuv',
    transactionDate: '2022-12-15T14:26:46.480Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'tfovH9wWzUf0EJz9',
    transactionDate: '2022-12-15T14:26:19.418Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'F2nvheJxpD7zHH73',
    transactionDate: '2022-12-15T14:24:47.901Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'GYqduO6UFxwFmblW',
    transactionDate: '2022-12-15T14:23:39.090Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'GX6YtSFgQKj3xpmp',
    transactionDate: '2022-12-15T02:55:25.173Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '02sXNSg1bhiCDFTH',
    transactionDate: '2022-12-15T02:41:23.774Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '2lucTOHpTzrUr7xS',
    transactionDate: '2022-12-15T02:41:23.675Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'Aue66vQCp4rylRTc',
    transactionDate: '2022-12-15T01:54:10.939Z',
    type: 'CREDIT',
    details: 'Cash Contest Cancelled',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'ceaFhsmwrqpDnfDf',
    transactionDate: '2022-12-14T14:14:39.223Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'cVZkDNkFonc4EgnB',
    transactionDate: '2022-12-14T14:13:29.461Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'xxd8MIk4BdL3ioWO',
    transactionDate: '2022-12-14T14:12:59.681Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'iKdMFMVoNac2qvPJ',
    transactionDate: '2022-12-14T03:08:34.386Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 37.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'RSEei5Qf7xToKj4W',
    transactionDate: '2022-12-14T03:06:16.244Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 62.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '0v9pIQryYV9tzyP5',
    transactionDate: '2022-12-14T02:56:23.058Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '6p21jy6r7vmlYfRd',
    transactionDate: '2022-12-14T02:35:25.399Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 100,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'rpMpMIqjTmisy5mC',
    transactionDate: '2022-12-13T18:15:59.397Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '5GNgOim11EwoE72P',
    transactionDate: '2022-12-13T18:15:17.299Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'u5bw5xb3yofe8dHI',
    transactionDate: '2022-12-13T16:49:33.175Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'as8siTak9srCiJuB',
    transactionDate: '2022-12-13T16:48:00.509Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'Tg6khxpGZGEaoOcL',
    transactionDate: '2022-12-11T23:21:27.144Z',
    type: 'DEBIT',
    details: 'Payout',
    amount: 162.5,
    amountBonus: 0,
    status: 'PENDING',
  },
  {
    id: 'avpnfaueTi9SpqdE',
    transactionDate: '2022-12-11T23:20:08.722Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'wnEYJW3rCBeq2bFp',
    transactionDate: '2022-12-10T21:37:04.429Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'ckJFULHMxhBAo9Wl',
    transactionDate: '2022-12-10T03:34:25.177Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'pAKvAPsfi6EIVMiZ',
    transactionDate: '2022-12-10T03:33:52.040Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'fmKzvSdfGRqdDddI',
    transactionDate: '2022-12-10T02:47:03.547Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'kkFN1WmVwhGZxGCL',
    transactionDate: '2022-12-09T17:57:33.554Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'AucqHTJxyXUouFnH',
    transactionDate: '2022-12-09T14:06:54.966Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'e7u7AuGtgm5cf1ja',
    transactionDate: '2022-12-09T14:06:46.103Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'o60g3KQxFM0OHTsP',
    transactionDate: '2022-12-09T04:32:03.380Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 25,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'orQBggOLKzZVE6Q0',
    transactionDate: '2022-12-09T00:40:43.872Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'mXQ2xDI251dqgE4g',
    transactionDate: '2022-12-09T00:39:41.345Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'fBuIt2gwqUQWLc02',
    transactionDate: '2022-12-07T14:47:12.823Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 38,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'OGbOjJUPmvJnT4pv',
    transactionDate: '2022-12-07T09:05:08.645Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 19,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'zmJUq3YTzosPCEQW',
    transactionDate: '2022-12-07T09:05:08.643Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 19,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'f1GRvNcCOPeiEgs1',
    transactionDate: '2022-12-07T02:21:33.852Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 100,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'Mx5zi54YFKQBmfAN',
    transactionDate: '2022-12-07T02:19:52.747Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 100,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'eRw7xrJ1xmhdEPFt',
    transactionDate: '2022-12-07T02:18:35.641Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 100,
    amountBonus: 0,
    status: 'FAILED',
  },
  {
    id: 'wviHMSU3Q3Rs8cSB',
    transactionDate: '2022-12-06T17:45:56.241Z',
    type: 'CREDIT',
    details: 'Free credit',
    amount: 0,
    amountBonus: 50,
    status: 'COMPLETE',
  },
  {
    id: 'Oxcr1Rl1pfVuKAJC',
    transactionDate: '2022-12-06T17:28:37.099Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 38,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'beJ3qmDGNRMNmTuo',
    transactionDate: '2022-12-06T17:25:44.101Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 100,
    amountBonus: 0,
    status: 'FAILED',
  },
  {
    id: '0f32k4TbLgp5dRq0',
    transactionDate: '2022-12-06T17:10:23.472Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 100,
    amountBonus: 0,
    status: 'FAILED',
  },
  {
    id: 't8zwgdu5qgVfK04n',
    transactionDate: '2022-12-06T06:01:48.396Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'uR7SiPVvXudhadph',
    transactionDate: '2022-12-06T05:55:03.389Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'GYQA6z2SRXkDkohh',
    transactionDate: '2022-12-06T05:50:57.065Z',
    type: 'CREDIT',
    details: 'Deposit',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'P6rND3o3W2RGsyWH',
    transactionDate: '2022-12-06T02:50:17.814Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 0.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'ol0Ew6cDZiGbGYRO',
    transactionDate: '2022-12-06T02:50:17.789Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 0.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'pq0cNkOvkTAyrAt9',
    transactionDate: '2022-12-06T02:50:17.749Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 0.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'HH3RRTYHmxBhWMTL',
    transactionDate: '2022-12-06T02:50:17.691Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 0.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'La8P3bhRm0T0U1kc',
    transactionDate: '2022-12-06T02:50:17.611Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 0.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'oUC21q1wB23c7vZD',
    transactionDate: '2022-12-06T02:50:17.587Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 0.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'J5gQ1qcwVOAKuDZ8',
    transactionDate: '2022-12-05T23:30:14.275Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 4,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'N1MQa04EIiZoTyVY',
    transactionDate: '2022-12-05T13:29:55.447Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 1,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'CvNX1SE5cXopgRf3',
    transactionDate: '2022-12-05T13:29:05.053Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'MaZUcaKi7XzUZmHk',
    transactionDate: '2022-12-05T07:22:53.527Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 10,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'Zl6gJbolkiM6hsW9',
    transactionDate: '2022-12-05T02:05:53.529Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'hVhegYdKjPAcffrG',
    transactionDate: '2022-12-05T01:40:45.853Z',
    type: 'DEBIT',
    details: 'Place entry',
    amount: 50,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: '6CPHaA9At5kfktbY',
    transactionDate: '2022-12-04T23:34:59.237Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'tEPuiCLmAJiow0rp',
    transactionDate: '2022-12-04T23:34:59.197Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'UlzBESiamxy40eUP',
    transactionDate: '2022-12-04T23:34:59.180Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'bVM4lHAeEf59Vo3o',
    transactionDate: '2022-12-04T23:34:59.175Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'sPygYCKjgeoeuY5T',
    transactionDate: '2022-12-04T23:34:58.999Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'fUJ05roMvyALxtgJ',
    transactionDate: '2022-12-04T23:34:58.957Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'sktr7ELZIfi0LLzK',
    transactionDate: '2022-12-04T23:34:58.955Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'AB5ohvKDkPw0nqwt',
    transactionDate: '2022-12-04T23:34:58.916Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
  {
    id: 'KqKk3UZyKi37OQ4m',
    transactionDate: '2022-12-04T23:34:58.837Z',
    type: 'CREDIT',
    details: 'Win',
    amount: 12.5,
    amountBonus: 0,
    status: 'COMPLETE',
  },
];
