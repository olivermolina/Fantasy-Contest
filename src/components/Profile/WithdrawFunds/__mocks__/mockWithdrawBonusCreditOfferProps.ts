export const mockWithdrawBonusCreditOfferProps = {
  handleSubmit: () => console.log('submit'),
  handleNext: () => console.log('next'),
  open: true,
  setOpen: () => console.log('set open'),
  maxRetentionBonus: 500,
  retentionBonusMatchMultiplier: 2,
  cashBalance: 100,
};
