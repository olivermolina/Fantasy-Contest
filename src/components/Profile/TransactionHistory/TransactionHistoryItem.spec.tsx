import React from 'react';
import { render } from '@testing-library/react';
import TransactionHistoryItem from './TransactionHistoryItem';
import { mockTransactions } from './__mocks__/mockTransactions';

describe('TransactionHistoryItem', () => {
  it('should render correctly', () => {
    const props = {
      row: mockTransactions[1]!,
    };

    const { container } = render(<TransactionHistoryItem {...props} />);
    expect(container).toMatchSnapshot();
  });
});
