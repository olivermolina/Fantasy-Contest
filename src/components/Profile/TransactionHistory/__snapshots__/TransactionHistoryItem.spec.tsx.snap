// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionHistoryItem should render correctly 1`] = `
<div>
  <div
    class="grid grid-cols-2 lg:grid-cols-6 gap-2 lg:gap-4 bg-inherit p-4"
  >
    <p
      class="flex flex-col gap-2"
    >
      <span
        class="text-lightText text-sm lg:text-md"
      >
        Transaction Date
      </span>
      <span
        class="text-white text-md lg:text-lg"
      >
        Dec 22, 2022 03:25 AM
      </span>
    </p>
    <p
      class="flex flex-col gap-2"
    >
      <span
        class="text-lightText text-sm lg:text-md"
      >
        Amount
      </span>
      <span
        class="text-white text-md lg:text-lg"
      >
        25.00
      </span>
    </p>
    <p
      class="flex flex-col gap-2"
    >
      <span
        class="text-lightText text-sm lg:text-md"
      >
        Type
      </span>
      <span
        class="text-white text-md lg:text-lg"
      >
        CREDIT
      </span>
    </p>
    <p
      class="flex flex-col gap-2"
    >
      <span
        class="text-lightText text-sm lg:text-md"
      >
        Free Credits
      </span>
      <span
        class="text-white text-md lg:text-lg"
      >
        0.00
      </span>
    </p>
    <p
      class="flex flex-col gap-2"
    >
      <span
        class="text-lightText text-sm lg:text-md"
      >
        Details
      </span>
      <span
        class="text-white text-md lg:text-lg"
      >
        Win
      </span>
    </p>
    <p
      class="flex flex-col gap-2"
    >
      <span
        class="text-lightText text-sm lg:text-md"
      >
        Status
      </span>
      <div
        class="flex w-fit text-md rounded-lg lg:text-lg bg-[#1A4265] items-center gap-2 p-2 "
      >
        <span
          class="rounded-full p-2 w-2 h-2 bg-green-500"
        />
        <p
          class="uppercase text-md text-white text-green-500"
        >
          COMPLETE
        </p>
      </div>
    </p>
  </div>
</div>
`;
